File: src/3015.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(countOfPairs(3, 1, 3))
}

func countOfPairs(n int, x int, y int) []int {
	res := make([]int, n+1)
	if y < x {
		x, y = y, x
	}
	x--
	y--

	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			minPath := min(j-i, Abs(i-x)+1+Abs(j-y))
			res[minPath] += 2
		}
	}

	return res[1:]
}

func Abs(num int) int {
	return int(math.Abs(float64(num)))
}

File: 3015.java
class Solution {
    public int[] countOfPairs(int n, int x, int y) {
        int[] res = new int[n];

        for (int i = 1; i <= n; ++i) {
            for (int j = i + 1; j <= n; ++j) {
                int direct = j - i;
                int viaXY = Math.abs(i - x) + 1 + Math.abs(j - y);
                int viaYX = Math.abs(i - y) + 1 + Math.abs(j - x);
                int d = Math.min(direct, Math.min(viaXY, viaYX));

                res[d - 1] += 2;
            }
        }
        return res;
    }
}